src/backend/storage/file/fd.c

```diff
@@ -1256,9 +1260,9 @@ OpenNamedTemporaryFile(const char *fileName,
 	 * force it into the database's default tablespace, so that it will not
 	 * pose a threat to possible tablespace drop attempts.
 	 */
-	if (numTempTableSpaces > 0 && !interXact && !GetForceDefaultTableSpaceVal())
+	if (!interXact && !GetForceDefaultTableSpaceVal())
```

Comment:
condition was changed, comments - not?

Reply:
This condition was unnecessary, since `GetNextTempTableSpace()` and `GetNextFileTempTableSpace()` return `InvalidOid`, if no tablespaces are set, and we are checking for `InvalidOid` anyway. I think adding `(numTempTableSpaces > 0 || fileNumTempTableSpaces > 0) && ...` to every condition is too clunky

Reply:
I still think you should update the comment above.

Reply:
Done

Reply:
comment above "If some temp tablespace(s) have been given to us, ..." was not updated!

Reply:
I added the comment above `if (OidIsValid(tblspcOid))`, which is slightly below this line. I think it the comment is better if it's closer to what it describes.